Firebase/Core/FIRApp.m

```diff
@@ -293,26 +301,39 @@ - (instancetype)initInstanceWithName:(NSString *)name options:(FIROptions *)opti
 - (BOOL)configureCore {
   [self checkExpectedBundleID];
   if (![self isAppIDValid]) {
-    if (_options.usingOptionsFromDefaultPlist && [self isDataCollectionDefaultEnabled]) {
+    if (_options.usingOptionsFromDefaultPlist && [self isDataCollectionDefaultEnabled] &&
+        NSClassFromString(@"FIRDiagnostics")) {
+      FIRDiagnosticsData *diagnosticsData = [[FIRDiagnosticsData alloc] init];
+      [diagnosticsData insertValueIfNotNil:@([self isDataCollectionDefaultEnabled])
```

Comment:
Since FIRDiagnosticsData knows about FIRApp, wouldn't it be cleaner if it collected those values (e.g. user agent, config type) from the inside, keeping FIRApp more concise?